internal/config/config.go

```diff
@@ -42,20 +42,48 @@ type Config struct {
 	Worker                 Worker         `yaml:"worker"`
 }
 
-func ParseFile(path string) (*Config, error) {
-	fd, err := os.Open(path)
-	if err != nil {
-		return nil, fmt.Errorf("could not open the configuration file: %v", err)
+func LoadConfigFromCM(cm *corev1.ConfigMap, cfg *Config) error {
+	const configKey = "controller_config.yaml"
+	rawConfig, ok := cm.Data[configKey]
+	if !ok {
+		return fmt.Errorf("key %q not found in ConfigMap %s/%s", configKey, cm.Namespace, cm.Name)
 	}
-	defer fd.Close()
+	return decodeStrictYAMLIntoConfig([]byte(rawConfig), cfg)
+}
 
-	cfg := Config{}
+func decodeStrictYAMLIntoConfig(yamlData []byte, config *Config) error {
+	decoder := yaml.NewDecoder(bytes.NewReader(yamlData))
+	decoder.KnownFields(true)
 
-	if err = yaml.NewDecoder(fd).Decode(&cfg); err != nil {
-		return nil, fmt.Errorf("could not decode configuration file: %v", err)
+	if err := decoder.Decode(config); err != nil {
+		return fmt.Errorf("error unmarshaling YAML: %w", err)
 	}
 
-	return &cfg, nil
+	return nil
+}
+
+func NewDefaultConfig() *Config {
+	runAsUser := int64(0)
+	firmwarePath := "/lib/firmware"
+
+	return &Config{
+		HealthProbeBindAddress: ":8081",
+		WebhookPort:            9443,
+		LeaderElection: LeaderElection{
+			Enabled:    true,
+			ResourceID: "kmm.sigs.x-k8s.io",
+		},
+		Metrics: Metrics{
+			EnableAuthnAuthz: true,
+			BindAddress:      "0.0.0.0:8443",
+			SecureServing:    true,
+		},
+		Worker: Worker{
+			RunAsUser:        &runAsUser,
+			SELinuxType:      "spc_t",
+			FirmwareHostPath: &firmwarePath,
```

Comment:
```suggestion
			FirmwareHostPath: ptr.To("/lib/firmware"),
```
And remove the variable.